ksp_plugin_adapter/main_window.cs

```diff
@@ -377,51 +374,72 @@ internal class MainWindow : SupervisedWindowRenderer {
   }
 
   private void RenderPredictionSettings() {
-    bool enabled = vessel_ != null;
     if (vessel_ != predicted_vessel_()) {
       vessel_ = predicted_vessel_();
-      string vessel_guid = vessel_?.id.ToString();
-      if (vessel_guid != null && plugin.HasVessel(vessel_guid)) {
-        enabled = true;
-        AdaptiveStepParameters adaptive_step_parameters =
-            plugin.VesselGetPredictionAdaptiveStepParameters(vessel_guid);
-        prediction_length_tolerance_index_ = Array.FindIndex(
-            prediction_length_tolerances_,
-            (double tolerance) =>
-                tolerance >=
-                    adaptive_step_parameters.length_integration_tolerance);
-        if (prediction_length_tolerance_index_ < 0) {
-          prediction_length_tolerance_index_ =
-              default_prediction_length_tolerance_index_;
-        }
-        prediction_steps_index_ = Array.FindIndex(
-            prediction_steps_,
-            (long step) => step >= adaptive_step_parameters.max_steps);
-        if (prediction_steps_index_ < 0) {
-          prediction_steps_index_ = default_prediction_steps_index_;
-        }
-      } else {
-        enabled = false;
+    }
+    bool enabled = false;
+    AdaptiveStepParameters? adaptive_step_parameters = null;
+    string vessel_guid = vessel_?.id.ToString();
+    if (vessel_guid != null && plugin.HasVessel(vessel_guid)) {
+      enabled = true;
+      adaptive_step_parameters =
+          plugin.VesselGetPredictionAdaptiveStepParameters(vessel_guid);
+      prediction_length_tolerance_index_ = Array.FindIndex(
+          prediction_length_tolerances_,
+          (double tolerance) =>
+              tolerance >=
+                  adaptive_step_parameters.Value.length_integration_tolerance);
+      if (prediction_length_tolerance_index_ < 0) {
+        prediction_length_tolerance_index_ =
+            default_prediction_length_tolerance_index_;
       }
+      prediction_steps_index_ = Array.FindIndex(
+          prediction_steps_,
+          (long step) => step >= adaptive_step_parameters.Value.max_steps);
+      if (prediction_steps_index_ < 0) {
+        prediction_steps_index_ = default_prediction_steps_index_;
+      }
+    } else {
+      enabled = false;
```

Comment:
This else clause has no effect.
Let us just get rid of `enabled`, and use `adaptive_step_parameters.HasValue` which is now equivalent below (we display them only if they are there).

Name it at the call site, i.e., `enabled : adaptive_step_parameters.HasValue`.
Consider reordering the arguments so as not to have to name all the other parameters (booleans benefit from naming more often than other parameters).

Reply:
Done.